api/db-cdr/prep-tables/make-prep-survey.py

```diff
@@ -0,0 +1,470 @@
+                            if len(split_codes) > 1:
+                                long_code_to_short_code[split_codes[0].strip().lower()] = \
+                                    split_codes[1].strip().lower()
+
+                # items that are marked as suppressed will be stored with the key as flag
+                # and the values as a list
+                answer_suppression = {}
+                for item in field_annotation_array:
+                    split = item.split('=')
+                    if "CONTROLLED_ANSWER_SUPPRESSED" in split:
+                        answer_suppression.setdefault("CONTROLLED_ANSWER_SUPPRESSED", [])
+                        answer_suppression["CONTROLLED_ANSWER_SUPPRESSED"].append(split[1].lower())
+
+                    if "REGISTERED_ANSWER_SUPPRESSED" in split:
+                        answer_suppression.setdefault("REGISTERED_ANSWER_SUPPRESSED", [])
+                        answer_suppression["REGISTERED_ANSWER_SUPPRESSED"].append(split[1].lower())
+
+                ############ Topic ############
+                if len(row['Section Header']) > 0:
+                    header = row['Section Header']
+
+                    # find the new lines and index to get the header up to the new line
+                    # to do: need to figure out what to do with first topic in per_med_his
+                    position = header.find("\n")
+                    if position == -1:
+                        header = row['Section Header'].replace("Thanks for your answers.", "")
+                    else:
+                        header = row['Section Header'][:position].replace("Thanks for your answers.", "")
+
+                    # get the concept codes for the topics and write topics into file
+                    topic_codes = query_topic_code(project, dataset, header.replace("'", "\\'"))
+                    topic_id_all = id_all
+                    topic_code = ''
+                    if not topic_codes:
+                        topic_code = topic_codes[0]
+                    write_row_all(all_writer, survey_id_all, topic_code, header, 'TOPIC', None, None, item_flags,
+                                  long_code_to_short_code)
+
+                    # in the case that we have topic_suppressed we want to skip that row
+                    if item_flags["CONTROLLED_TOPIC_SUPPRESSED"] != 1:
+                        topic_id_controlled = id_controlled
+                        write_row(controlled_writer, survey_id_controlled, topic_code, header
+                                  , 'TOPIC', None, None, item_flags, 'controlled', long_code_to_short_code)
+
+                    if item_flags["REGISTERED_TOPIC_SUPPRESSED"] != 1:
+                        topic_id_register = id_registered
+                        write_row(registered_writer, survey_id_registered, topic_code, header
+                                  , 'TOPIC', None, None, item_flags, 'register', long_code_to_short_code)
```

Comment:
Your type flags are not consistent. If you look at line 112, you used "registered" but here you use "register". Pick one